src/idl_gen_cpp.cpp

```diff
@@ -895,16 +195,692 @@ class CppGenerator : public BaseGenerator {
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      auto &field = **it;
+      if (it != struct_def.fields.vec.begin()) code += ", ";
+      code += field.name + "_(";
+      if (IsScalar(field.value.type.base_type)) {
+        code += "flatbuffers::EndianScalar(";
+        code += GenUnderlyingCast(field, false, "_" + field.name);
+        code += "))";
+      } else {
+        code += "_" + field.name + ")";
+      }
+      GenPadding(field, code, padding_id, PaddingInitializer);
+    }
+
+    code += " {";
+    padding_id = 0;
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      auto &field = **it;
+      GenPadding(field, code, padding_id, PaddingDeclaration);
+    }
+    code += " }\n\n";
+
+    // Generate accessor methods of the form:
+    // type name() const { return flatbuffers::EndianScalar(name_); }
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      auto &field = **it;
+      GenComment(field.doc_comment, code_ptr, nullptr, "  ");
+      auto is_scalar = IsScalar(field.value.type.base_type);
+      code += "  " +
+              GenTypeGet(field.value.type, " ", "const ", " &", true);
+      code += field.name + "() const { return ";
+      code += GenUnderlyingCast(
+          field, true, is_scalar
+                           ? "flatbuffers::EndianScalar(" + field.name + "_)"
+                           : field.name + "_");
+      code += "; }\n";
+      if (parser_.opts.mutable_buffer) {
+        if (is_scalar) {
+          code += "  void mutate_" + field.name + "(";
+          code += GenTypeBasic(field.value.type, true);
+          code += " _" + field.name + ") { flatbuffers::WriteScalar(&";
+          code += field.name + "_, ";
+          code += GenUnderlyingCast(field, false, "_" + field.name);
+          code += "); }\n";
+        } else {
+          code += "  ";
+          code += GenTypeGet(field.value.type, "", "", " &", true);
+          code += "mutable_" + field.name + "() { return " + field.name;
+          code += "_; }\n";
+        }
+      }
+    }
+    code += "};\nSTRUCT_END(" + struct_def.name + ", ";
+    code += NumToString(struct_def.bytesize) + ");\n\n";
+  }
+
+  // Set up the correct namespace. Only open a namespace if
+  // the existing one is different (opening/closing only what is necessary)
+  // the file must start and end with an empty namespace
+  void CheckNameSpace(const Namespace *ns, std::string *code_ptr) {
+    if (cur_name_space_ == ns) return;
+    auto s1 = cur_name_space_ == nullptr ? 0 : cur_name_space_->components.size();
+    auto s2 = ns == nullptr ? 0 : ns->components.size();
+    std::vector<std::string>::size_type limit = 0;
+    while (limit < s1 && limit < s2 &&
+           ns->components[limit] == cur_name_space_->components[limit])
+      limit++;
+    for (auto j = s1; j > limit; --j)
```

Comment:
limit can be equal to 0 and it happens :  at the start and the end of the C++ file, it is supposed to be equal to 0.

As we always have 0 <= limit <= s1 and 0<= limit <= s2p, 
the conditions j = s1 and  j > limit ensures that we always have --j >= 0 in the for loop